rtv/content.py

```diff
@@ -367,52 +385,96 @@ def __init__(self, name, submissions, loader, order=None):
             raise exceptions.SubredditError('No submissions')
 
     @classmethod
-    def from_name(cls, reddit, name, loader, order=None, query=None):
+    def from_name(cls, reddit, name, loader, order=None, query=None,
+                  listing='r', period=None):
+
+        # Strip leading, trailing, and redundant backslashes
+        n = ''
+        n = ''.join([n + ''.join(list(g)) if k != '/' else '/' \
+                                       for k, g in groupby(name)]).strip(' /')
```

Comment:
Not sure what this is doing but it looks like it could be simplified, or at least expanded so it's easier to read. What is ``groupby()`` doing?